### ISSUE TITLE
New loss function for segmentation

### ISSUE THREAD

**Issue by fepegar** [NifTK/NiftyNet#200]
Hi all,

I have implemented the [Tversky loss function](https://arxiv.org/abs/1706.05721) for segmentation with imbalanced data. I'll try it tomorrow and if it works I'll open a PR.

It needs two parameters α and β to weight precision and recall during training (if α + β = 1, it's equivalent to an Fβ-score and if α = β = 0.5, it's Dice). Is there a way to specify those without hard-coding them? The only other losses that accept kwargs are GDSC and `wasserstein_disagreement_map`, but I the only example I've found in the repo in which those are passed is in this test:
https://github.com/NifTK/NiftyNet/blob/0bdbd9cafefc8063110e1152f29d5001ad8e6906/tests/loss_segmentation_test.py#L94-L110

**Comment by wyli** [comment]
Thanks @fepegar, the loss function's inferface is ready, but we need the new configuration file https://github.com/NifTK/NiftyNet/issues/166  to support a proper nested parameter parsing.

Perhaps @BostonChildrensRadiology would be interested in this implementation as well.

**Comment by fepegar** [comment]
Cool, looking forward to the new config format.

I'll submit the PR as a WIP in a second.

**Comment by amartya-k** [comment]
Is this working for class imbalanced semantic segmentation?(mostly medical images)

**Comment by fepegar** [comment]
@amartya-k yes, it should. Take a look at the paper I referenced in the first comment.
